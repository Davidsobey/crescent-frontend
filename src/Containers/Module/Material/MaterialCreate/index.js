// do a file upload
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Field, reduxForm } from 'redux-form';
import { MenuItem } from 'material-ui/Menu';

import Select from '../../../../Components/Select';
import Card from '../../../../Components/Card';
import Button from '../../../../Components/Button';
import ModuleActions from '../../../../Actions/ModuleActions';
import LinearProgress from '../../../../Components/LinearProgress';

const validate = () => {
  const errors = {};

  return errors;
};

class MaterialCreate extends React.Component {
  constructor(props) {
    super(props);
    this.props.dispatch(ModuleActions.getAll());
  }

  state = {
    selectedFile: null,
  };

  fileSelectedHandler = (event) => {
    this.setState({
      selectedFile: event.target.files[0],
    });
  }

  submit = (values) => {
    const file = this.state.selectedFile;
    this.props.dispatch(ModuleActions.uploadMaterial(
      values.ModuleId,
      file,
    ));
  };

  render() {
    return ( 
      <Card width="600px" title="Create New Material">
        <form
          onSubmit={this.props.handleSubmit(this.submit)}
          noValidate
          autoComplete="off"
        >
          <div>
            <div className="width200">
              {this.props.modules ? (
                <Field name="ModuleId" label="Module Name" component={Select}>
                  {this.props.modules.map(module => (
                    <MenuItem value={module.id} key={module.id}>
                      {module.name}
                    </MenuItem>
                  ))}
                </Field>
              ) : (
                <div>
                  <LinearProgress color="secondary" />
                    Loading Modules
                </div>
              )}
              <div>
                <input type="file" onChange={this.fileSelectedHandler} />
              </div>
            </div>
          </div>
          <div className="alignRight">
            <Button variant="raised" color="primary" type="submit">
              Create Material
            </Button>
          </div>
        </form>
      </Card>
    );
  }
}

MaterialCreate.propTypes = {
  dispatch: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  modules: PropTypes.array,
};

const mapStateToProps = state => ({
  modules: state.ModuleReducer.modules,
});

const withForm = reduxForm(
  {
    form: 'materialCreate',
    validate,
  },
  MaterialCreate,
);

export default compose(connect(mapStateToProps), withForm)(MaterialCreate);
