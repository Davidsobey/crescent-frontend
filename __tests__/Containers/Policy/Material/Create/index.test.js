import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import MaterialCreate from '../../../../../src/Containers/Policy/Material/Create/index.js';


const mockStore = configureStore();
const initialState = {
  PolicyReducer:{
    policies: [],
    loading: false,
    newPolicyId: 1,
    creating: false,
  },
};
const store = mockStore(initialState);

describe('<PolicyMaterialCreate />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<MaterialCreate store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});