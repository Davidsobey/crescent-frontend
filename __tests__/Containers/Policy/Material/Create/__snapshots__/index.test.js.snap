// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PolicyMaterialCreate /> render() renders the component 1`] = `
<Connect(Form(PolicyMaterialCreate))
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="policyMaterialCreate"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  material_creating={false}
  newPolicyId={1}
  persistentSubmitErrors={false}
  policies={Array []}
  policies_loading={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;
